### PR TITLE
Feature/edit profile to database

### PR DESCRIPTION
**Pull request by sandroquist**
This pull request introduces several enhancements and new features to the `BeatLink` app, focusing on profile management, dependency updates, and permission handling. The most important changes include the addition of new dependencies, updates to profile-related classes, and modifications to handle new permissions for accessing media.

### Profile Management Enhancements:
* Introduced `ProfileRepositoryFirestore` to handle fetching and updating user profiles from Firestore, replacing the previous `ProfileRepository`. 
* Updated `ProfileViewModel` to use the new `ProfileRepositoryFirestore` and added methods to update profile data and handle profile picture URLs.
* Modified `ProfileData` to include a `Uri` for the profile picture and a list of favorite music genres.

### UI and Testing Updates:
* Added `ProfilePicture` composable to display profile pictures using Coil in `MainComponents.kt`.
* Updated `EditProfileScreen` to use the `ProfileViewModel` and added a mock URI for testing purposes in `EditProfileTest.kt`. 

### Permission Handling:
* Added new permissions for accessing media images in `AndroidManifest.xml` to support devices running Android 12L and higher.

These changes collectively enhance the app's profile management capabilities, streamline dependency management, and ensure proper handling of media permissions.

Note: The implementation isn't the most optimal and has a little issue, I tried to fix it as I could but not completely. This is why you can find some unused or commented functions, they will be used at a later date.

### CODE REVIEW

**Review by lunasrc — CHANGES_REQUESTED** [review]
Great job on this PR! The new features brings solid improvements to the app's profile management functionality. The code is well-documented, making it clear and easy to understand. Before merging, there is one small suggestion to make the implementation even stronger.

File: app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt [hunk 1]
```diff
@@ -17,35 +26,67 @@ import androidx.compose.material3.HorizontalDivider
...
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
 @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun EditProfileScreen(navigationActions: NavigationActions) {
-  var name by remember { mutableStateOf("This is the current name") }
+fun EditProfileScreen(profileViewModel: ProfileViewModel, navigationActions: NavigationActions) {
+  LaunchedEffect(Unit) { profileViewModel.fetchProfile() }
+  val profileData by profileViewModel.profile.collectAsState()
+  val maxDescriptionLength = 100
+  val maxUsernameLength = 20
```
[the remark this answers is not in the document]
Comment (lunasrc) [reply app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt:59]: Not ProfileData.kt, not ideal oops

[the remark this answers is not in the document]
Comment (sandroquist) [reply app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt]: good idea, I will put those variables inside of `ProfileData` as `companion object`

**Review by lunasrc — APPROVED** [review]
Great ! The use of companion objects for constants is a nice touch, keeping the code organized and maintainable. Now it is all set for merging. Well done !

**Pull request merged**